### ISSUE TITLE
Streaming doesn't work correctly with caching

### ISSUE THREAD

**Issue by xsyann** [langchain-ai/langchain#9762]
### System Info

LangChain version : 0.0.273
Python version : 3.10.8
Platform : macOS 13.5.1

### Information

- [ ] The official example notebooks/scripts
- [X] My own modified scripts

### Related Components

- [X] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [ ] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [X] Callbacks/Tracing
- [ ] Async

### Reproduction

```python
import langchain
from langchain.chat_models import ChatOpenAI
from langchain.callbacks.streaming_stdout import StreamingStdOutCallbackHandler
from langchain.cache import InMemoryCache


langchain.llm_cache = InMemoryCache()

llm = ChatOpenAI(model_name="gpt-3.5-turbo", streaming=True, callbacks=[StreamingStdOutCallbackHandler()])
resp = llm.predict("Tell me a joke")
resp1 = llm.predict("Tell me a joke")
```

output:

```
Sure, here's a classic one for you:

Why don't scientists trust atoms?

Because they make up everything!
```

### Expected behavior

I'd expect both responses to be streamed to stdout but as the second one is coming from the memory cache, the callback handler `on_llm_new_token` is never called and thus the second response never printed.

I guess `on_llm_new_token` should be called once with the full response loaded from cache to ensure a consistent behavior between new and cached responses.

**Comment by auxon** [comment]
I made a bunch of fixes for Lanarky for this to work.  See https://github.com/auxon/lanarky/commit/b5904a0a58e508f227c5f3da5f6ce06758509b54, if it helps.

**Comment by hinthornw** [comment]
Does the `llm.stream()` work for your use case?


![image](https://github.com/langchain-ai/langchain/assets/13333726/4d31bbb6-8c94-4d61-96cb-6f421d4df7ef)

**Comment by EliiseS** [comment]
Does anybody have a working sample on how to do caching and streaming with OpenAI? I've looked at the code of the streaming implementation and it doesn't look like it contains any caching calls.

There's also an open ticket on this issue within the JS SDK: [Caching not implemented for OpenAI LLM stream usage · Issue #3582 · langchain-ai/langchainjs](https://github.com/langchain-ai/langchainjs/issues/3582)

**Comment by saitaiky** [comment]
Just wondering if there’s been any update on this behaviour? Currently, when a response is served from the cache, it doesn’t get streamed back. My expectation would be that the `messages` stream event still gets emitted, with the cached value returned as the response.

When building a conversational application, receiving only the `updates ` stream makes it difficult to distinguish between a status update and an actual message response (even from cache). This ambiguity complicates the logic for handling streamed interactions.

It would be great if the stream behaviour could be consistent regardless of whether the response is cached or freshly generated.
